bundles/org.openhab.binding.mqtt.homeassistant/src/main/java/org/openhab/binding/mqtt/homeassistant/internal/component/Lock.java

```diff
@@ -95,27 +108,33 @@ public Lock(ComponentFactory.ComponentConfiguration componentConfiguration) {
                     return true;
                 }).build();
 
-        String[] commands;
-        if (channelConfiguration.payloadOpen == null) {
-            commands = new String[] { channelConfiguration.payloadLock, channelConfiguration.payloadUnlock, };
-        } else {
-            commands = new String[] { channelConfiguration.payloadLock, channelConfiguration.payloadUnlock,
-                    channelConfiguration.payloadOpen };
+        Map<String, String> commands = new HashMap<>();
```

Comment:
Setting up the mapping for lock states is done right here. From the constant "LOCK" to custom payload.